code/game/machinery/computer/pod.dm

```diff
@@ -1,21 +1,38 @@
 /obj/machinery/computer/pod
 	name = "mass driver launch control"
 	desc = "A combined blastdoor and mass driver control unit."
+	/// Connected mass driver
 	var/obj/machinery/mass_driver/connected = null
-	var/title = "Mass Driver Controls"
+	/// ID of the launch control
 	var/id = 1
-	var/timing = 0
+	/// If the launch timer counts down
+	var/timing = FALSE
+	/// Time before auto launch
 	var/time = 30
+	/// Range in which we search for a mass drivers and poddoors nearby
 	var/range = 4
 
-
 /obj/machinery/computer/pod/Initialize()
 	. = ..()
 	for(var/obj/machinery/mass_driver/M in range(range, src))
 		if(M.id == id)
 			connected = M
+			break
 
+/obj/machinery/computer/pod/process()
```

Comment:
Consider setting a subsystem_type var on this machine alongside using begin_processing and end_processing procs for machines, or START_PROCESSING and STOP_PROCESSING macros.

You want the machine to be processing only when necessary and to stop processing when there's no timers active, nothing's counting down, you're not waiting for anything.